x/staking/client/cli/tx.go

```diff
@@ -298,24 +228,24 @@ func BuildCreateValidatorMsg(cliCtx context.CLIContext, txBldr authtxb.TxBuilder
 	if delAddr != "" {
 		delAddr, err := sdk.AccAddressFromBech32(delAddr)
 		if err != nil {
-			return txBldr, nil, err
+			return nil, err
 		}
 
 		msg = staking.NewMsgCreateValidatorOnBehalfOf(
-			delAddr, sdk.ValAddress(valAddr), pk, amount, description, commissionMsg,
+			delAddr, cliCtx.FromValAddr(), pk, amount, description, commissionMsg,
 		)
 	} else {
 		msg = staking.NewMsgCreateValidator(
-			sdk.ValAddress(valAddr), pk, amount, description, commissionMsg,
+			cliCtx.FromValAddr(), pk, amount, description, commissionMsg,
 		)
 	}
 
-	if viper.GetBool(FlagGenesisFormat) {
+	if viper.GetBool(client.FlagGenerateOnly) {
```

Comment:
This requires thorough testing of `gaiad gentx`

Reply:
What additional testing is required? I'm pretty sure this behavior is caught in tests.

Reply:
Going to resolve this comment.